### DISCUSSION TITLE
How to make inference on InnerEye-DeepLearning models deployed locally?

### DISCUSSION THREAD

**Discussion by Med-Rokaimi** [microsoft/InnerEye-DeepLearning#819] [Q&A] [answered]
Hey,
I deployed the InnerEye deeplearning locally on Linux OS.
I can see there are many h5 models. Plus I trained a [sample model ](https://innereye-deeplearning.readthedocs.io/md/sample_tasks.html).
But, I cannot run inference on the existing models.
I tried this command:
<pre>
python InnerEye/Scripts/submit_for_inference.py --image_file  Tests/ML/test_data/test_img.nii.gz --model=GlaucomaPublic
</pre> 
It requires an Azure subscription while I wanna run the toolkit locally.

<pre>
File "/home/ai/InnerEye-DeepLearning/InnerEye/Azure/azure_config.py", line 205, in get_workspace
    raise ValueError("The values for 'subscription_id' and 'resource_group' were not found. "
ValueError: The values for 'subscription_id' and 'resource_group' were not found. Was the Azure setup completed?

</pre>
Any ideas how to run inferences on InnerEye deeplearning models will be highly appreciated.
Note: I didn't use InnerEye-inference.

**Comment by peterhessey** [comment, marked as answer]
Hi! Apologies for the delayed response on this. 

The `submit_for_inference.py` script is not for running inference locally, it is only for submitting inference jobs to AzureML, hence the prompting for `subscription_id` and `resource_group` settings.

To run inference locally, see the ["Testing an existing model"](https://innereye-deeplearning.readthedocs.io/md/building_models.html?highlight=from%20local%20checkpoints#from-local-checkpoints) docs page. This shows you how to use the `--no-train` and `--local_weights_path` flags to run your model locally in testing mode.

Note that this will run inference on all images in your test set. You can further control which / how many images are passed through the model through the use of the `--restrict_subjects` flag, as in the [debugging and monitoring docs page](https://innereye-deeplearning.readthedocs.io/md/debugging_and_monitoring.html#shorten-training-run-time-for-debugging).

**Reply by Med-Rokaimi** [reply]
Thank you so much. worked.
May I ask please, if are there pre-trained models that could be included in the toolkit?
I can see there are a few segmentation and classification models included when installing InnerEye-deeplearning toolkit such as Lung, Prostate, Glaucoma. 
But I can see very interesting models for cancer detection and so on your videos on Youtube and the project page. 
I'm wondering where to find more trained models available to be included locally.
Thank you so much.

**Reply by peterhessey** [reply]
Glad to hear it 😄 

We are currently in the process of releasing a Lung segmentation model. Keep an eye on the repo for a release over the next few days!
